### PR TITLE
Migrate to new sqlcipher package

### PR DESCRIPTION
**Pull request by raphaelm**
Google play marks the SQL cipher package as an outdated SDK and might block app releases in the future. This PR [does the migration](https://www.zetetic.net/sqlcipher/sqlcipher-for-android-migration/) and unfortunately needs to vendor some requery code for that.

### CODE REVIEW

**Comment by robbi5** [comment]
This is also an upgrade from the sqlcipher major version 3 to 4. 

https://discuss.zetetic.net/t/upgrading-to-sqlcipher-4/3283 says:
> However, the introduction of new default algorithms, increased KDF iterations and a larger page size means that SQLCipher 4 will not open older databases by default.

We should do the `PRAGMA cipher_migrate;` call, or am I missing some place where it happens implicitly?

**Comment by robbi5** [comment]
Well, we don't do a automatic migration, but it looks like we kill and rebuild the database:

https://github.com/pretix/pretixscan-android/blob/master/pretixscan/app/src/main/java/eu/pretix/pretixscan/droid/PretixScan.kt#L54-L58

**Comment by raphaelm** [comment]
ouch! how did i not notice that. no, that's not he goal xD

**Comment by raphaelm** [comment]
Oh wow, they could really have mentioned this or given better guidance :D Fixed!

File: pretixscan/app/src/main/java/eu/pretix/pretixscan/droid/PretixScan.kt [hunk 1]
```diff
@@ -31,6 +34,23 @@ class PretixScan : MultiDexApplication() {
...
+
+        System.loadLibrary("sqlcipher")
+
+        val databaseFile = getDatabasePath(Models.DEFAULT.name)
+        SQLiteDatabase.openOrCreateDatabase(databaseFile, dbPass, null, null, object: SQLiteDatabaseHook {
+            override fun preKey(connection: SQLiteConnection) {
+            }
+
+            override fun postKey(connection: SQLiteConnection) {
+                connection.execute("PRAGMA cipher_migrate;", emptyArray(), null)
```
Comment (robbi5) [inline pretixscan/app/src/main/java/eu/pretix/pretixscan/droid/PretixScan.kt:49]:
```suggestion
                val result = connection.executeForLong("PRAGMA cipher_migrate;", emptyArray(), null)
                if (result != 0L) {
                    throw SQLiteException("cipher_migrate failed")
                }
```

> android.database.sqlite.SQLiteException: unknown error (code 0): Queries can be performed using SQLiteDatabase query or rawQuery methods only.

See https://github.com/sqlcipher/sqlcipher-android/issues/6 and https://github.com/sqlcipher/sqlcipher-android/blob/master/sqlcipher/src/androidTest/java/net/zetetic/database/sqlcipher_cts/MigrateDatabaseFrom1xFormatToCurrentFormatTest.java#L28 - we should check the return value here

**Pull request merged**